packages/api/src/api/contributions.ts

```diff
@@ -0,0 +1,233 @@
+import express, { Router } from 'express';
+import { env } from '../env';
+import { Contribution } from '../entities/Contribution';
+import { User } from '../entities/User';
+import logger from '../logger';
+import { Project } from '../entities/Project';
+
+export const contributions = Router();
+contributions.use(express.json());
+
+export interface PullRequestContribution {
+  id: string;
+  title: string;
+  permalink: string;
+  mergedAt: string;
+  author: {
+    login: string;
+  };
+}
+
+export interface Repository {
+  id: string;
+  nameWithOwner: string;
+}
+
+interface ContributionResponse {
+  data: {
+    search: {
+      pageInfo: {
+        hasNextPage: boolean;
+        endCursor: string | null;
+      };
+      nodes: PullRequestContribution[];
+    };
+  };
+}
+
+interface RepositoryResponse {
+  data: {
+    nodes: Repository[];
+  };
+}
+
+const buildProjectsQuery = async (projects: Project[]) => {
+  let idString = '';
+
+  projects.forEach((p) => {
+    idString += `"${p.nodeID}",`;
+  });
+
+  idString = idString.substring(0, idString.length - 1);
+
+  const repoQueryString = `
+    query { 
+      nodes(ids: [${idString}]){
+        ... on Repository {
+          id
+          nameWithOwner
+        }    
+      }
+    }
+  `;
+
+  const fetchRes = await fetch('https://api.github.com/graphql', {
```

Comment:
We need a try catch around this and the call to `.json` below

Reply:
done